docs/designs/ruby_test_plan_002.md

```diff
@@ -0,0 +1,116 @@
+       Given  clean repository of a "ill-defined project"
+        When  a user request to fetch dependencies for a "ill-defined project"
+        Then  Cachi2 reports an error
+         And  repository for "ill-defined project" remains unchanged
+    Examples:
+        |    ill-defined project                           |
+        | Gemfile.lock is missing                          |
+        | Git dependency is not pinned to a revision       |
+        | Transitive dependency is ill-defined             |
+
+
+@integration_test
+Feature: Cachi2 can preset configuration for hermetic builds
+
+    Scenario outline: a user prepares a repository with a Ruby project
+        When  a user generates environment for "a Ruby project"
+        Then  the environment contains a pointer to local Bundler config
+         And  Bundler cache directory structure is recreated for every project
+    Examples:
+        |    a Ruby project                                       |
+        | Single-project repository with mixed depenendcies types |
+        | Multi-project repository with mixed depenendcies types  |
+
+    Scenario: a user tries to prepare a project with malformed path dependency
+        When  a user generates environment for "malformed path dependency project"
+        Then  Cachi2 reports an error
+         And  repository for "malformed path dependency project" remains unchanged
+
+    Rule: fetched dependencies directory structure must replicate Bundler cache structure
+        Example: a user prepares simple Ruby project with just gem dependencies
+            When a user generates an environment for "purely gem-dependent project"
+            Then dependencies are stored in vendor/cache
+             And .gem dependencies are kept as is
+
+        Example: a user prepares simple Ruby project with git dependencies
+            When a user generates an environment for "git and gem-dependent project"
+            Then dependencies are stored in vendor/cache
+             And Git dependnecies directories  names contain revisions
+
+        Example: a user prepares complex Ruby project
+            When a user generates an environment for "complex Ruby project"
+            Then dependencies are stored in vendor/cache
+             And dependencies are not duplicated
+
+
+### Necessary repositories
+
+Following the existing practice just one repository is needed. This repository
+must contain branches representing test scenarios. I.e. each Ruby project
+described below is a branch in the test repository.
+
+Each line below describes a properly defined Ruby project:
+ 1.  with a mixture of gem, git and path dependencies and a .gemspec file;
+ 2.  with a mixture of gem, git and path dependencies without a .gemspec file;
+
+These repositories must transitively depend on other gems, their depednencies must
+contain common dependencies.
+
+Each line below describes a malformed Ruby project:
```

Comment:
Missing Gemfile is a problem also, if for nothing else then bundler parser complains about it whenever git dependencies are used as it's trying to figure out the cache path where the git repo should be cloned to, but I think it would complain even harder during install.

Reply:
This is a good call-out, thank you, and looks like a candidate for poorly-formed repository scenario. I'll add that scenario back.